lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -1137,9 +1134,67 @@ def test_coach_tab_for_ccx_advance_settings(self, ccx_feature_flag, enable_ccx,
             course = self.ccx_enabled_course if enable_ccx else self.ccx_disabled_course
             self.assertEquals(
                 expected_result,
-                self.check_ccx_tab(course)
+                self.check_ccx_tab(course, self.user)
             )
 
+    def test_ccx_tab_visibility_for_staff_when_not_coach_master_course(self):
+        """
+        Staff cannot view ccx coach dashboard on master course by default.
+        """
+        staff = self.make_staff()
+        self.assertFalse(self.check_ccx_tab(self.course, staff))
+
+    def test_ccx_tab_visibility_for_staff_when_coach_master_course(self):
+        """
+        Staff can view ccx coach dashboard only if he is coach on master course.
+        """
+        staff = self.make_staff()
+        role = CourseCcxCoachRole(self.course.id)
+        role.add_users(staff)
+        self.assertTrue(self.check_ccx_tab(self.course, staff))
+
+    def test_ccx_tab_visibility_for_staff_ccx_course(self):
+        """
+        Staff can access coach dashboard on ccx course.
+        """
+        staff = self.make_staff()
+        self.make_coach()
+        ccx = self.make_ccx()
+        ccx_key = CCXLocator.from_course_locator(self.course.id, unicode(ccx.id))
+
+        with ccx_course(ccx_key) as course_ccx:
+            allow_access(course_ccx, staff, 'staff')
+            self.assertTrue(self.check_ccx_tab(course_ccx, staff))
+
+    def test_ccx_tab_visibility_for_instructor_when_not_coach_master_course(self):
+        """
+        Instructor cannot view ccx coach dashboard on master course by default.
+        """
+        instructor = self.make_instructor()
+        self.assertFalse(self.check_ccx_tab(self.course, instructor))
+
+    def test_ccx_tab_visibility_for_instructor_when_coach_master_course(self):
+        """
+        Instructor can view ccx coach dashboard only if he is coach on master course.
+        """
+        instructor = self.make_instructor()
+        role = CourseCcxCoachRole(self.course.id)
+        role.add_users(instructor)
+        self.assertTrue(self.check_ccx_tab(self.course, instructor))
+
+    def test_ccx_tab_visibility_for_instructor_ccx_course(self):
+        """
+        Instructor can access coach dashboard on ccx course.
+        """
+        instructor = self.make_instructor()
+        self.make_coach()
+        ccx = self.make_ccx()
+        ccx_key = CCXLocator.from_course_locator(self.course.id, unicode(ccx.id))
+
+        with ccx_course(ccx_key) as course_ccx:
+            allow_access(course_ccx, instructor, 'instructor')
+            self.assertTrue(self.check_ccx_tab(course_ccx, instructor))
+
```

Comment:
There are already tests for coach, you cannot see them in diff, but if you
expand file you can see those tests.
On Jul 12, 2016 8:21 PM, "Giovanni Di Milia" <notifications@github.com>
wrote:

> In lms/djangoapps/ccx/tests/test_views.py
> <https://github.com/mitocw/edx-platform/pull/250#discussion_r70460794>:
>
> > +        role.add_users(instructor)
> > +        self.assertTrue(self.check_ccx_tab(self.course, instructor))
> > +
> > +    def test_ccx_tab_visibility_for_instructor_ccx_course(self):
> > +        """
> > +        Instructor can access coach dashboard on ccx course.
> > +        """
> > +        instructor = self.make_instructor()
> > +        self.make_coach()
> > +        ccx = self.make_ccx()
> > +        ccx_key = CCXLocator.from_course_locator(self.course.id, unicode(ccx.id))
> > +
> > +        with ccx_course(ccx_key) as course_ccx:
> > +            allow_access(course_ccx, instructor, 'instructor')
> > +            self.assertTrue(self.check_ccx_tab(course_ccx, instructor))
> > +
>
> Can you add an extra test just for the coach to assert that she sees the
> tab even if she is not staff or instructor?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mitocw/edx-platform/pull/250/files/a3714da6b18237ae9793ba973b9260d507753251#r70460794>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ8rEtjfTEMsClDsMwjCkx2xQ94FzveUks5qU7ETgaJpZM4I_Gaq>
> .
>